### ISSUE TITLE
Two REST API calls inside withAPIData?

### ISSUE THREAD

**Issue by mrleemon** [WordPress/gutenberg#6528]
## Issue Overview

I'm developing a dynamic block that retrieves post data using the REST API. I'm wondering if one can make two REST API calls at the same time inside withAPIData:

```
registerBlockType( 'my-plugin/my-post-block', {
    title: 'My Post Block',
    icon: 'megaphone',
    category: 'widgets',
    attributes: {
        post: {
            type: 'string',
            default: '0',
        },
    },

    edit: withAPIData( ( props ) => {
        return {
            posts: '/wp/v2/posts',
            singlepost: `/wp/v2/posts/${ props.attributes.post }`
        };
    } )( ( props ) => {
        if ( ! props.posts.data ) {
            return "loading !";
        }
        if ( ! props.singlepost.data ) {
            return "loading !";
        }
        if ( props.singlepost.data === '0' ) {
            return "No post selected";
        }

        ....

    } )
} );
```

Right now I'm getting a permanent "loading !" message instead of data from both endpoints. Is this supported? Do I have to make two separate REST API calls? If so, how?

Thanks in advance

**Comment by jasonbahl** [comment]
You could use WPGraphQL to request multiple resources in a single request. 😀

**Comment by theleemon** [comment]
Is this related maybe? #6277

**Comment by theleemon** [comment]
I installed 2.8 and the issue is mitigated somehow. Now, from time to time, one of the REST API calls stalls indefinitely. Hope that the #6277 fix gets into 2.9.

**Comment by mrleemon** [comment]
Updated to 2.9 and I'm still having problems.

**Comment by mrleemon** [comment]
Tried again with GB 3.0.1 and the issue still persists. Right now, it's impossible to call two endpoints one after another without the risk of one of them failing. I hope you can find a fix for #6303 soon.

**Comment by mrleemon** [comment]
As `withAPIData` will be deprecated in favor of `withSelect` in the near future, I'm closing this issue.
